### PR TITLE
[FEATURE] LEDGER_AUTH_{BASIC,HTTP_SIGNATURE}_ENABLED

### LINKED ISSUE #86: Ability to turn on/off auth schemes
It would be good to turn off HTTP Basic auth entirely via the configuration.

`LEDGER_AUTH_BASIC_ENABLED=0` - Turn off Basic auth
`LEDGER_AUTH_HTTP_SIGNATURE_ENABLED=0` - Turn off http-signature based auth

### PR DESCRIPTION
**Pull request by sentientwaffle**
Fixes #86.

### CODE REVIEW

**Comment by MatthewPhinney** [comment]
LGTM

File: src/services/config.js [hunk 1]
```diff
@@ -27,3 +27,8 @@ if (admin_pass) {
     pass: admin_pass
   }
 }
+
+config.auth_enabled = {
```
Comment (clark800) [inline src/services/config.js]: `auth_enabled` kind of sounds like it should be a boolean

Comment (sentientwaffle) [inline src/services/config.js]: The values are. They are configured as strings because all env variables are strings.

Comment (clark800) [inline src/services/config.js]: Right, I just meant that in the code "config.auth_enabled" would look like a boolean, but I'm not sure if there is a better option. "config.auth.basic_enabled" is one other option. Not a big deal though, not trying to block the merge, just an idea.

Comment (justmoon) [inline src/services/config.js]: Agree with Chris - `config.auth.basic_enabled` matches our (not super-strict, but let's make an effort) config naming scheme better: `config.[module].[option]`

**Comment by sentientwaffle** [comment]
Ok, renamed.

**Comment by sentientwaffle** [comment]
Ok, if the env variable is undefined, it will default to true. If its `'0'`, it is false. Anything else will exit.

File: src/services/config.js [hunk 2]
```diff
@@ -27,3 +27,15 @@ if (admin_pass) {
...
 }
+
+config.auth = {
+  basic_enabled: getBoolean('AUTH_BASIC_ENABLED'),
+  http_signature_enabled: getBoolean('AUTH_HTTP_SIGNATURE_ENABLED')
+}
+
+function getBoolean (env) {
+  const val = config.getEnv(env)
+  if (val === undefined) return true
```
Comment (clark800) [inline src/services/config.js]: Should probably allow "1" here also

**Comment by sentientwaffle** [comment]
Ok, `"1"` -> `true`

**Comment by clark800** [comment]
LGTM

**Pull request merged**